graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/user/TopSecondDegreeByCountForUser.java

```diff
@@ -45,6 +46,13 @@ public TopSecondDegreeByCountForUser(
     StatsReceiver statsReceiver) {
     super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
   }
+  protected boolean isValidNodeInfoUpdate(
+      TopSecondDegreeByCountRequestForUser request,
+      EdgeIterator edgeIterator) {
+    long keepEdgeWithinTime = request.getKeepSocialProofsWithinTime();
+    long edgeLastEngagedTime = ((TimestampEdgeIterator)edgeIterator).getCurrentEdgeEngagementTime();
```

Comment:
edgeLastEngagedTime => edgeEngagementTime; the term last seems to suggest there are multiple engagement times for one edge.